src/main/java/uk/ac/cam/cl/dtg/isaac/quiz/IsaacNumericValidator.java

```diff
@@ -495,4 +495,29 @@ private String reformatNumberForParsing(final String numberToFormat) {
         reformattedNumber = reformattedNumber.replaceFirst(BARE_POWER_OF_TEN_REGEX, "1e${exp1}${exp2}");
         return reformattedNumber;
     }
+
+    /**
+     *  Replace explanation of validation response if question has default feedback and existing feedback blank or generic.
+     *
+     *  This method could be void, since it modifies the object passed in by reference, but it makes for shorter and
+     *  simpler code when it is used if it just returns the same object it is passed.
+     *
+     * @param question - the question to use the default feedback of
+     * @param response - the response to modify if necessary
+     * @return the modified response object
+     */
+
+    private QuantityValidationResponse useDefaultFeedbackIfNecessary(final IsaacNumericQuestion question,
+                                                                     final QuantityValidationResponse response) {
+        Content feedback = response.getExplanation();
+        boolean feedbackEmptyOrGeneric = feedbackIsNullOrEmpty(feedback) || DEFAULT_VALIDATION_RESPONSE.equals(feedback.getValue())
+                || DEFAULT_WRONG_UNIT_VALIDATION_RESPONSE.equals(feedback.getValue());
+
+        if (null != question.getDefaultFeedback() && feedbackEmptyOrGeneric) {
+            log.debug("Replacing generic or blank explanation with default feedback from question.");
+            response.setExplanation(question.getDefaultFeedback());
+            // TODO - should this preserve the 'sig_figs' tag? If so, how to do it without modifying the referenced default feedback?
```

Comment:
Yes, the problem is there is no obvious way to get a mapper into this class. Presumably Guice would work it all out if I made a custom constructor and annotated it `@Inject` with the mapper? We never instatiate this class by hand, only using `injector.getInstance(...)` in the `QuestionManager`.

As for the message, I worry it will vary based on context, but yes; it is a content thing.

Reply:
Cool, if we use injector.getInstance(...) then yes I believe Guice will just do what the nice thing 👍

Reply:
Great. I might make a card for this to be considered, but for now I think we don't make it more complex than necessary.